packages/common/src/writeContract.ts

```diff
@@ -63,43 +71,59 @@ export async function writeContract<
     queueConcurrency: opts.queueConcurrency,
   });
 
-  async function prepareWrite(): Promise<
-    WriteContractParameters<abi, functionName, args, chain, account, chainOverride>
-  > {
+  const feeRef = getFeeRef({ client: opts.publicClient ?? client, refreshInterval: 10000 });
+
+  async function prepare(): Promise<WriteContractParameters<abi, functionName, args, chain, account, chainOverride>> {
     if (request.gas) {
-      debug("gas provided, skipping simulate", request.functionName, request.address);
+      debug("gas provided, skipping preparation", request.functionName, request.address);
       return request;
     }
 
-    debug("simulating", request.functionName, "at", request.address);
-    const result = await simulateContract<chain, account | undefined, abi, functionName, args, chainOverride>(
-      opts.publicClient ?? client,
+    const { abi, address, args, dataSuffix, functionName } = request;
+    const data = encodeFunctionData({
+      abi,
+      args,
+      functionName,
+    } as EncodeFunctionDataParameters);
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { nonce, maxFeePerGas, maxPriorityFeePerGas, ...preparedTransaction } = await prepareTransactionRequest(
+      client,
       {
+        // The nonce and fee values don't need to be accurate for gas estimation
+        // and we can save a couple rpc calls by providing stubs here
+        nonce: 0,
+        maxFeePerGas: 0n,
+        maxPriorityFeePerGas: 0n,
+        ...defaultParameters,
         ...request,
-        blockTag: "pending",
```

Comment:
why did we lose the pending block tag?

Reply:
discussed offline, this is useful for estimating against other txs in the mempool, bringing it back